CMakeLists.txt

```diff
@@ -50,65 +55,64 @@ else()
     set_target_properties(docopt_o PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
 
     add_library(docopt SHARED $<TARGET_OBJECTS:docopt_o>)
-	set_target_properties(docopt PROPERTIES
-			VERSION ${PROJECT_VERSION}
-			SOVERSION ${PROJECT_VERSION_MAJOR}
-			)
+    set_target_properties(docopt PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR})
     add_library(docopt_s STATIC $<TARGET_OBJECTS:docopt_o>)
 endif()
 
 target_include_directories(docopt PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}> $<INSTALL_INTERFACE:include/docopt>)
 target_include_directories(docopt_s PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}> $<INSTALL_INTERFACE:include/docopt>)
 
+add_library(docopt::docopt ALIAS docopt)
+add_library(docopt::docopt_static ALIAS docopt_s)
+
 if(MSVC)
     # DOCOPT_DLL: Must be specified when building *and* when using the DLL.
     #             That's what the "PUBLIC" means.
     # DOCOPT_EXPORTS: Must use __declspec(dllexport) when building the DLL.
     #                 "PRIVATE" means it's only defined when building the DLL.
-    target_compile_definitions(docopt PUBLIC  DOCOPT_DLL
-                                      PRIVATE DOCOPT_EXPORTS)
+    target_compile_definitions(
+        docopt
+        PUBLIC DOCOPT_DLL
+        PRIVATE DOCOPT_EXPORTS
+    )
 endif()
 
 if(NOT MSVC)
-	set_target_properties(docopt PROPERTIES OUTPUT_NAME docopt)
-	set_target_properties(docopt_s PROPERTIES OUTPUT_NAME docopt)
+    set_target_properties(docopt PROPERTIES OUTPUT_NAME docopt)
+    set_target_properties(docopt_s PROPERTIES OUTPUT_NAME docopt)
 endif()
 
 if(USE_BOOST_REGEX)
-	add_definitions("-DDOCTOPT_USE_BOOST_REGEX")
-	# This is needed on Linux, where linking a static library into docopt.so
-	# fails because boost static libs are not compiled with -fPIC
-	set(Boost_USE_STATIC_LIBS OFF)
-    find_package(Boost 1.53 REQUIRED COMPONENTS regex)
-    include_directories(${Boost_INCLUDE_DIRS})
-    target_link_libraries(docopt ${Boost_LIBRARIES})
-	if(WITH_STATIC)
-		target_link_libraries(docopt_s ${Boost_LIBRARIES})
-	endif()
+    # This is needed on Linux, where linking a static library into docopt.so
```

Comment:
This is a good modernization, switching from add_definitions to target_compile_definitions on specific projects